### PR TITLE
feat: add `multi-bundle` command

### PR DESCRIPTION
**Pull request by zamotany**

### CODE REVIEW

File: packages/haul-basic-bundle-webpack-plugin/src/WebpackBasicBundlePlugin.ts [hunk 1]
```diff
@@ -15,7 +15,23 @@ function asyncEval(url) {
  * Adds React Native specific tweaks to bootstrap logic.
  */
 export default class WebpackBasicBundlePlugin {
-  constructor(private bundle: boolean, private sourceMap: boolean) {}
+  bundle: boolean;
+  sourceMap: boolean;
+  preloadBundles: string[];
```
Comment (thymikee) [inline packages/haul-basic-bundle-webpack-plugin/src/WebpackBasicBundlePlugin.ts]: private?

File: packages/haul-basic-bundle-webpack-plugin/src/WebpackBasicBundlePlugin.ts [hunk 2]
```diff
@@ -98,8 +114,13 @@ export default class WebpackBasicBundlePlugin {
         (compilation.mainTemplate as any).hooks.bootstrap.tap(
           'WebpackBasicBundlePlugin',
           (source: string) => {
-            // throw new Error(source);
-            return `${asyncEval}\n${source}`;
+            const preload = this.preloadBundles.length
+              ? `${this.preloadBundles.map(
+                  bundleName =>
+                    `this.bundleRegistryLoad("${bundleName}", true, true)\n`
```
Comment (thymikee) [inline packages/haul-basic-bundle-webpack-plugin/src/WebpackBasicBundlePlugin.ts]:
```suggestion
                    `this.bundleRegistryLoad("${bundleName}", true, true);\n`
```

File: packages/haul-cli/src/commands/multiBundle.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+import {
+  Runtime,
+  getProjectConfigPath,
+  getNormalizedProjectConfigBuilder,
+  NormalizedProjectConfig,
+} from '@haul-bundler/core';
+import * as messages from '../messages/multiBundleMessages';
+import SimpleProgressWebpackPlugin from 'simple-progress-webpack-plugin';
+
+export default function ramBundleCommand(runtime: Runtime) {
```
Comment (dratwas) [inline packages/haul-cli/src/commands/multiBundle.ts]: Shouldn't it be called `multiBundleCommand`?

Comment (thymikee) [inline packages/haul-cli/src/commands/multiBundle.ts]:
```suggestion
export default function multiBundleCommand(runtime: Runtime) {
```

File: packages/haul-cli/src/commands/multiBundle.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+          dev,
+          minify,
+          platform,
+          assetsDest,
+          sourcemapOutput,
+          progress,
+        } = argv;
+
+        // TODO: figure out a better way to read and transpile user files on-demand
+        require('@haul-bundler/core-legacy/build/babelRegister');
```
Comment (thymikee) [inline packages/haul-cli/src/commands/multiBundle.ts]: possibly could be moved to `getNormalizedProjectConfigBuilder ` now? (in a separate PR)

File: packages/haul-cli/src/commands/multiBundle.ts (hunk 3 above)
Comment (zamotany) [reply packages/haul-cli/src/commands/multiBundle.ts]: fixed

File: packages/haul-cli/src/commands/multiBundle.ts [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+          }
+        }
+
+        runtime.logger.done(
+          'All bundles successfully compiled. You can now run your React Native multi-bundle app.'
+        );
+        runtime.complete();
+      } catch (error) {
+        runtime.logger.error(error.message);
+        runtime.logger.error(error.stack);
```
Comment (thymikee) [inline packages/haul-cli/src/commands/multiBundle.ts]: the message is usually in the stack, isn't it? unless removed explicitly

File: packages/haul-core/src/config/types.ts [hunk 6]
```diff
@@ -36,7 +36,8 @@ export type BundleConfig = {
     Exclude<keyof MinifyOptions, 'sourceMap'>
   >;
   sourceMap?: boolean | 'inline';
-  dllDependencies?: string[];
+  dll?: boolean;
+  dependsOn?: string[];
```
Comment (thymikee) [inline packages/haul-core/src/config/types.ts]: don't forget to update the docs

Comment (zamotany) [reply packages/haul-core/src/config/types.ts]: the docs needs to be redone from scratch, so I'll do it at the end of this project

**Pull request merged**